### ISSUE TITLE
Type annotations

### ISSUE THREAD

**Issue by marvinvanaalst** [numba/llvmlite#828]
## Feature request: type annotations

First, thanks for the great work you folks have done here!

I've transitioned most of my Python code onto using type annotations and I much enjoy having them checked statically. Is there any interest in the community here for that (or at least in the common APIs, such as `IRBuilder`)?

I'd volunteer to go through the chore of adding them in if folks are interested.

Cheers

**Comment by sklam** [comment]
Yes, we would like to enable typing annotations. It will likely help users catch certain kind of mistakes early. Contributions are welcomed.

**Comment by marvinvanaalst** [comment]
Quick nudge: has anyone had time to take a look at the proposed code? :)
